Source/Core/Core/AchievementManager.cpp

```diff
@@ -322,6 +352,48 @@ bool AchievementManager::IsHardcoreModeActive() const
   return rc_client_is_processing_required(m_client);
 }
 
+void AchievementManager::FilterApprovedPatches(std::vector<PatchEngine::Patch>& patches,
+                                               const std::string& game_ini_id) const
+{
+  if (!IsHardcoreModeActive())
+    return;
+
+  if (!m_ini_root.contains(game_ini_id))
+    patches.clear();
+  auto patch_itr = patches.begin();
+  while (patch_itr < patches.end())
```

Comment:
`patch_itr != patches.end()` is the more correct way to check for reaching the end of an iterator.